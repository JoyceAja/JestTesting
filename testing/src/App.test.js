import React from 'react';
import ReactDOM from 'react-dom';

import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
import App from './App';

describe('App (Snapshot)', () => {
  it('App renders input and button', () => {
    // const tree = renderer.create(<App />) //why do we need this??
    // const json = tree.toJSON()
    const wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot()
  })
})

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('should display a text input to fill question', () =>{
  const wrapper = shallow(<App />);
  expect(wrapper.find('input[type="text"]').length).toEqual(1);
});

it('should display a submit button to send question to the Internet Gods', () =>{
  const wrapper = shallow(<App />);
  expect(wrapper.find('button[type="submit"]').length).toEqual(1);
} )