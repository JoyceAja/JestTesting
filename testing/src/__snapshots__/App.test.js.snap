// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App (Snapshot) App renders input and button 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="encloser"
>
        <input
                type="text"
        />
        <button
                type="submit"
        >
                Ask the God's of the internet
        </button>
</div>,
      "className": "App",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            type="text"
/>,
          <button
            type="submit"
>
            Ask the God's of the internet
</button>,
        ],
        "className": "encloser",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Ask the God's of the internet",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Ask the God's of the internet",
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="encloser"
>
          <input
                    type="text"
          />
          <button
                    type="submit"
          >
                    Ask the God's of the internet
          </button>
</div>,
        "className": "App",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              type="text"
/>,
            <button
              type="submit"
>
              Ask the God's of the internet
</button>,
          ],
          "className": "encloser",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Ask the God's of the internet",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Ask the God's of the internet",
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
